src/main/java/FlowerStore/ItemManager.java

```diff
@@ -18,48 +31,85 @@ public void removeItemById(Integer id) {
 
     }
 
+    public List<Item> getItems() {
+        return items;
+    }
+
+
     @Override
     public String toString() {
         return "ItemManager\n\n" +
                 "items:\n" + items.toString();
     }
-
     public void sortByField(String field, SortOrder order){
-        switch (field){
-            case "itemName":
-                if (order.equals(SortOrder.ASC)) {
-                    Collections.sort(this.items, (i1, i2) -> (i1.getItemName().compareTo(i2.getItemName())));
-                    System.out.println("Sorted by itemName (ASC)");
-                } else if (order.equals(SortOrder.DESC)) {
-                    Collections.sort(this.items, (i1, i2) -> (i1.getItemName().compareTo(i2.getItemName())));
-                    Collections.reverse(this.items);
-                    System.out.println("Sorted by itemName (DESC)");
+        Comparator<Item> comparator = LIST_COMPARATOR.get(field);
+        if (order.equals(SortOrder.ASC)){
+            Collections.sort(this.items, comparator);
+        } else if (order.equals(SortOrder.DESC)){
+            Collections.sort(this.items, comparator);
+            Collections.reverse(this.items);
+        }
+    }
+
+    public List<BouquetFlower> compileBouquet(BouquetConfiguration configuration, List<Item> base){
+
+        // define configurations
+
+        List<BouquetFlower> baseFlowers = base.stream()
+                .filter(item -> item instanceof BouquetFlower)
+                .map(item -> (BouquetFlower) item)
+                .collect(Collectors.toList());
+        List<BouquetFlower> bouquet= new ArrayList<>();
+        String color = configuration.getColor();
+        int lowerBound = configuration.getPriceRange().getLowerBound();
+        int upperBound = configuration.getPriceRange().getUpperBound();
+        List<String> flower = configuration.getFlowers();
+        int count = configuration.getCount();
+        int flowerTypeCount = count / flower.size();
+        int priceTotal = 0;
+
+        // logic
+
+        for (String flowerType : flower) {
+            int i = 0;
+            for (BouquetFlower tempFlower : baseFlowers){
+                if (tempFlower.getItemName() == flowerType && tempFlower.getColor() == color && i < flowerTypeCount) {
```

Comment:
In case of such complex booleans, it makes sense to introduce a variable that has reasonable name